src/ua/sumdu/j2se/Potapenko/tasks/ArrayTaskList.java

```diff
@@ -0,0 +1,76 @@
+package ua.sumdu.j2se.Potapenko.tasks;
+
+public class ArrayTaskList{
+    Task[] task = new Task[0];
+    public void add(Task task) {
+        int count = this.task.length;
+        Task[] task1 = new Task[count + 1];
+        System.arraycopy(this.task, 0, task1, 0, count);
+        task1[count] = task;
+        this.task = null;
+        this.task = task1;
```

Comment:
No need to do `this.task = null` when you are overriding the array link on the next line